source/common/upstream/upstream_impl.cc

```diff
@@ -679,16 +679,18 @@ ClusterImplBase::resolveProtoAddress(const envoy::api::v2::core::Address& addres
 
 ClusterInfoImpl::ResourceManagers::ResourceManagers(const envoy::api::v2::Cluster& config,
                                                     Runtime::Loader& runtime,
-                                                    const std::string& cluster_name) {
-  managers_[enumToInt(ResourcePriority::Default)] =
-      load(config, runtime, cluster_name, envoy::api::v2::core::RoutingPriority::DEFAULT);
+                                                    const std::string& cluster_name,
+                                                    Stats::Scope& stats_scope) {
+  managers_[enumToInt(ResourcePriority::Default)] = load(
+      config, runtime, cluster_name, stats_scope, envoy::api::v2::core::RoutingPriority::DEFAULT);
   managers_[enumToInt(ResourcePriority::High)] =
-      load(config, runtime, cluster_name, envoy::api::v2::core::RoutingPriority::HIGH);
+      load(config, runtime, cluster_name, stats_scope, envoy::api::v2::core::RoutingPriority::HIGH);
 }
 
 ResourceManagerImplPtr
 ClusterInfoImpl::ResourceManagers::load(const envoy::api::v2::Cluster& config,
                                         Runtime::Loader& runtime, const std::string& cluster_name,
+                                        Stats::Scope& stats_scope,
```

Comment:
Instead of passing the scope here, is there any way you can just pass the cluster stats? Then you don't have a potential naming inconsistency issue?